### PR TITLE
Patch EC2 Workflow

### PR DESCRIPTION
**Pull request by zjleblanc**
# General

Create a patching workflow that snapshots an EC2, applies patches, and restores from a snapshot if the patch fails. 

## Playbooks
- snapshot_ec2
- patch_ec2
- restore_ec2

## Workflow
- Cloud / AWS / Patch Workflow
Screenshot 2023-06-20 at 4 24 44 PM

### CODE REVIEW

**Comment by willtome** [comment]
@zjleblanc is this a duplicate of #74 ?

**Comment by zjleblanc** [comment]
@willtome yes, I went ahead and closed the other one.

**Comment by willtome** [comment]
Thanks @zjleblanc. Lets try reusing the existing patch jobs in the workflow. See example starting here https://github.com/ansible/product-demos/blob/81f35e8d67bd6971ae0f75ad0f7ec222cd757daa/satellite/setup.yml#L196

**Comment by zjleblanc** [comment]
@willtome updated this PR to use the existing linux patching job template. I added it to the cloud setup, not sure if there is a better way to reference the controller_config object while ensuring it's created without having to run the full linux setup.

File: cloud/setup.yml [hunk 1]
```diff
@@ -367,6 +367,91 @@ controller_templates:
...
           required: true
 
+  - name: Cloud / AWS / Snapshot EC2
+    job_type: run
+    organization: Default
+    credentials:
+      - AWS
+    project: Ansible official demo project
+    playbook: cloud/snapshot_ec2.yml
+    inventory: Workshop Inventory
```
Comment (willtome) [inline cloud/setup.yml:377]: Change to Demo Inventory

File: cloud/setup.yml [hunk 2]
```diff
@@ -367,6 +367,91 @@ controller_templates:
...
+          required: false
+
+  - name: Cloud / AWS / Restore EC2 from Snapshot
+    job_type: run
+    organization: Default
+    credentials:
+      - AWS
+    project: Ansible official demo project
+    playbook: cloud/restore_ec2.yml
+    inventory: Workshop Inventory
```
Comment (willtome) [inline cloud/setup.yml:408]: Change to Demo Inventory

File: cloud/setup.yml [hunk 3]
```diff
@@ -367,6 +367,91 @@ controller_templates:
...
+            - us-west-1
+            - us-west-2
+        - question_name: Specify target hosts
+          type: text
+          variable: _hosts
+          required: false
+
+  - name: "LINUX / Patching"
+    job_type: check
+    inventory: "Workshop Inventory"
```
Comment (willtome) [inline cloud/setup.yml:434]: Change to Demo Inventory

File: cloud/setup.yml [hunk 4]
```diff
@@ -367,6 +367,91 @@ controller_templates:
...
+    project: "Ansible official demo project"
+    playbook: "linux/patching.yml"
+    execution_environment: Default execution environment
+    notification_templates_started: Telemetry
+    notification_templates_success: Telemetry
+    notification_templates_error: Telemetry
+    use_fact_cache: true
+    ask_job_type_on_launch: true
+    credentials:
+      - "Workshop Credential"
```
Comment (willtome) [inline cloud/setup.yml:444]: Change to Demo Credential

**Review by willtome — CHANGES_REQUESTED** [review]

**Comment by willtome** [comment]
does this PR also resolve https://github.com/ansible/product-demos/issues/83? Something additional to test

**Pull request merged**